libopenage/util/file.cpp

```diff
@@ -92,5 +92,32 @@ std::vector<std::string> file_get_lines(const std::string &file_name) {
 	return result;
 }
 
+std::unordered_map<std::string, std::vector<std::string>> csv_file_map;
+
+void load_csv_files(std::string path) {
+
+	log::log(MSG(info) << "Loading csv files");
+	if (file_size(path) == -1) {
+		log::log(MSG(info) << "Cannot find csv file at " << path);
+		return;
+	}
+
+	std::vector<std::string> lines = file_get_lines(path);
+
+	std::string current_file = "";
+	for (auto& line : lines) {
+		if (line[0] == '#' && line[1] == '#' && line[2] == ' ') {
+			current_file = "./assets/converted" + line.erase(0, 3);
+			csv_file_map.emplace(current_file, std::vector<std::string>());
+		}
```

Comment:
This method should rather return the map that is currently global. Then the returned map is passed to the csv reader as pointer, which is `nullptr` by default. If it's not `nullptr`, the map is used as cache.

That is, the entry call for loading the csv files (`recurse_data_files`) gets this map as new argument.

The `./assets/converted` should also not be hardcoded. The prefix is already known as game spec basedir (which is `gamedata_dir` in `game_spec.cpp`).